public/components/create_report/create_report.tsx

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import {
+    EuiButtonEmpty,
+    EuiDatePicker,
+    EuiSelect,
+    EuiFlexGroup,
+    EuiFlexItem,
+    EuiFormRow,
+    EuiButton,
+    EuiPage,
+    EuiTitle,
+    EuiPageBody,
+    EuiSpacer,
+    EuiRadioGroup,
+  } from '@elastic/eui';
+import { htmlIdGenerator } from '@elastic/eui/lib/services';
+import moment, { Moment } from 'moment';
+import { ReportSettings } from './report_settings/report_settings';
+import { ReportDelivery } from './delivery/delivery';
+import { ReportSchedule } from './schedule/schedule';
+
+const idPrefix = htmlIdGenerator()();
+
+export function CreateReport() {
+  const [reportSettingsDashboard, setReportSettingsDashboard] = useState('');
+  const [deliveryEmailSubject, setDeliveryEmailSubject] = useState('');
+  const [deliveryEmailBody, setDeliveryEmailBody] = useState('');
+  const [scheduleRadioFutureDateSelected, setScheduleRadioFutureDateSelected] = useState(false);
+  const [scheduleRadioRecurringSelected, setScheduleRadioRecurringSelected] = useState(false);
+  const [scheduleUTCOffset, setScheduleUTCOffset] = useState(0);
+  const [scheduleFutureDate, setScheduleFutureDate] = useState(moment());
+  const [scheduleRecurringFrequency, setScheduleRecurringFrequency] = useState('Daily');
+  const [scheduleRecurringUTCOffset, setScheduleRecurringUTCOffset] = useState(0);
+  const [scheduleRecurringWeeklyDayOfWeek, setScheduleRecurringWeeklyDayOfWeek] = useState('Monday');
+  const [scheduleRecurringStartDate, setScheduleRecurringStartDate] = useState(moment());
+  const [scheduleRadioIdSelected, setScheduleRadioIdSelected] = useState(`${idPrefix}7`);
```

Comment:
Use can also use `useState` with object. Such as 

```
const reportState = useState<CreateReportState>({
....otherstates.
})
```

Reply:
Would I be defining the other states in `CreateReportState`? Otherwise I don't think I fully understand the added benefit

Reply:
Correct you can create a State which contains all the required info and move this inside une useState. to simplify